components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/FxaClient.kt

```diff
@@ -0,0 +1,423 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.appservices.fxaclient
+
+import android.util.Log
+import kotlin.coroutines.CoroutineContext
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.withContext
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import mozilla.appservices.sync15.DeviceType
+
+// This allows `adb logcat | grep fxa_client` to pick up all log items from this component
+internal const val LOG_TAG = "fxa_client"
+
+/**
+ * Firefox account client
+ */
+class FxaClient private constructor(
+    private val inner: FirefoxAccount,
+    private val persistCallback: FxaPersistCallback,
+    private val eventHandler: FxaEventHandler,
```

Comment:
Note that this creates a reference cycle between `FxaAccountManager`, `FxaClient`, and `FxaHandler` that goes through the Rust code and therefore will never be broken.  I think this is okay for now, since all of those objects are intended to live forever, but hopefully we can add some sort of weak reference support to side-step this.